examples/aes-example.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include <mococrw/symmetric_crypto.h>
+#include <iostream>
+
+using namespace mococrw;
+
+static const std::vector<uint8_t> plaintext = utility::fromHex("deadbeef");
+static const SymmetricCipherMode aeOperationMode = SymmetricCipherMode::GCM;
+static const SymmetricCipherMode plainOperationMode = SymmetricCipherMode::CBC;
+static const SymmetricCipherPadding padding = SymmetricCipherPadding::PKCS;
+static const std::vector<uint8_t> aesKey = utility::fromHex("000102030405060708090a0b0c0d0e0f");
+static const std::vector<uint8_t> associatedData = utility::fromHex("beefdead");
+static const std::vector<uint8_t> iv = utility::fromHex("00102030405060708090a0b0c0d0e0f0");
+static const size_t authTagLength = 16;
+
+struct AesAuthenticatedEncryptionResult {
+    std::vector<uint8_t> iv;
+    std::vector<uint8_t> ciphertext;
+    std::vector<uint8_t> authTag;
+};
+
+struct AesEncryptionResult {
+    std::vector<uint8_t> iv;
+    std::vector<uint8_t> ciphertext;
+};
+
+AesAuthenticatedEncryptionResult aesAuthenticatedEncryption()
+{
+    auto encryptorBuilder = AESCipherBuilder{aeOperationMode, SymmetricCipherKeySize::S_128, aesKey}
+            // Padding mode is optional. Default is SymmetricCipherPadding::PKCS
+            .setPadding(padding)
+            // The IV is optional. Default is a random value (Recommendation: use a random value)
+            .setIV(iv);
+    // the authTagLength is an optional value. Default 16 bytes. Only needed of AES AEAD.
```

Comment:
```suggestion
    // the authTagLength is an optional value. Default 16 bytes. Only needed by AEAD modes of operation.
```

I think this should be "needed by", not "needed of", and I'd reformulate to "AEAD modes of operation", since technically you could also do this with other algorithms (we just happen to usually use AES).